### PR TITLE
Allow eCommerce event to accept symbol hash keys

### PR DESCRIPTION
**Pull request by mscwilson**
Solves issue #128 where the validation for Items and Transactions for eCommerce events didn't allow for symbols as hash keys.

### CODE REVIEW

**Comment by mscwilson** [comment]
I branched off the rubocop branch rather than the release branch apparently...

**Comment by paulboocock** [comment]
If you squash and merge the rubocop branch into your release branch, you'll probably be better off cherry-picking you commit from the PR into release rather than trying to merge this.

File: snowplow-tracker.gemspec [hunk 1]
```diff
@@ -33,6 +33,6 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.1'
 
   s.add_runtime_dependency 'contracts', '~> 0.7', '< 0.17'
-  s.add_development_dependency 'rspec', '~> 2.14.1'
-  s.add_development_dependency 'webmock', '~> 1.17.4'
+  s.add_development_dependency 'rspec', '>= 2.0'
+  s.add_development_dependency 'webmock', '>= 1.17', '< 3.9'
```
Comment (paulboocock) [inline snowplow-tracker.gemspec:37]: This change has crept into this commit. Might end up with a conflict when merge/cherrypicking, just a heads up. Potentially merge the loosening PR into release after cherrypicking this one will work around it.

**Review by paulboocock — APPROVED** [review]

**Comment by paulboocock** [comment]
I see that its coveralls thats failed but might wanna rerun the Github Action before cherry-picking/merging.

**Comment by mscwilson** [comment]
Thanks, I've cherry picked as you suggested, so closing this.

**Pull request closed without merging**